pkg/controllers/capr/managesystemagent/managesystemagent.go

```diff
@@ -35,22 +36,43 @@ import (
 )
 
 const (
-	generationSecretName = "system-agent-upgrade-generation"
+	generationSecretName                  = "system-agent-upgrade-generation"
+	upgradeAPIVersion                     = "upgrade.cattle.io/v1"
+	upgradeDigestAnnotation               = "upgrade.cattle.io/digest"
+	systemAgentUpgraderServiceAccountName = "system-agent-upgrader"
 )
 
-var Kubernetes125 = semver.MustParse("v1.25.0")
+var (
+	Kubernetes125 = semver.MustParse("v1.25.0")
+
+	// GH5551FixedVersions is a slice of rke2 versions
+	// which have resolved GH-5551 for Windows nodes.
+	// ref:  https://github.com/rancher/rke2/issues/5551
+	// The SUC should not deploy plans to Windows nodes
+	// running a version less than the below for each minor.
+	// This check can be removed when 1.31.x is the lowest supported
+	// rke2 version.
+	GH5551FixedVersions = map[int]*semver.Version{
```

Comment:
Does this issue affect K3s clusters? The current behavior is that the plan is not being delivered to K3s clusters — is this expected? ( I think it should be ok but want to confirm)

Reply:
K3s does not support windows nodes, so there is not harm in not checking the distribution type. While it doesnt really impact anything, shouldn't be delivering windows plans to k3s clusters at all. That's something that we should address after this PR

Reply:
SGTM.

Reply:
I had the same thought lol